tools/scripts/generic_variables.mk

```diff
@@ -0,0 +1,24 @@
+# Create variables that will be used within src.mk
+
+# This variables will be set to their default value if not set by the user
+# It assumes that the makefile if invoked from No-OS/project/some_project
+
+# User must set PLATFORM variable between:
+# PLATFORM = xilinx
+# PLATFORM = altera
+# PLATFORM = aducm3029
+
+PROJECT			?= $(realpath .)
+TARGET 			?= $(notdir $(realpath .))
```

Comment:
Would it be a good idea here to derive the target from the `PROJECT` variable? In that way a user can only set the `PROJECT` variable.

Reply:
linux.mk and windows.mk use the TARGET variable